### PR TITLE
Fixes and Additions in EVS v2.0 #1 feature/EVSv2revision1

### PR DESCRIPTION
**Pull request by Ho-ChunHuang-NOAA**
## Pull Request Testing ##
For AQM EVSv2.0 requirement modification:
- Define $NET prior to $HOMEevs; apply this change across all J-jobs
- Change subprocess.Popen to subprocess.run in EVS v2.0 (if they are used at all).
- Remove old concatenating log file sections from portions of EVS that use MET/METplus (because LOG_TO_TERMINAL_ONLY = TRUE now). Keep concatenating in plotting.
- Any use of LOOP_ORDER can safely be removed across all components.
- Update config files, where necessary, to use beta versions without new warnings
- Use INITDATE instead of VDATE in prep
- Replace DCOMIN with DCOMINobs and DCOMINmodel variable names instead

- [ ] Describe testing already performed for this Pull Request:</br>
Test AQM prep for 20240808 and 20240809
Test AQM stats for 20240808 and 20240809
Test AQM plots for 20240808

Check logs file for any err=1 and warning message, all log file look good.
Check graphics in https://www.emc.ncep.noaa.gov/users/verification/air_quality/aqm/dev
 
- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

(1) Run AQM prep
(2) Point COMIN= to /lfs/h1/ops/prod/com/evs/v1.0/prep/aqm/
Run AQM stats

(3) Point COMIN= to /lfs/h1/ops/prod/com/evs/v1.0/stats/aqm/
Run AQM plots

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**
YES
- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**
YES
- [ ] Please complete this pull request review by **[Fill in date]**.</br>
N/A
## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by malloryprow** [comment]
Code changes look good! I'm going to get your branch checked out and start getting prep to run.

**Comment by malloryprow** [comment]
jevs_aqm_prep.sh

**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr513/EVS/dev/drivers/scripts/prep/aqm/jevs_aqm_prep_00.o148271017
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr513/evs/v2.0
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_aqm_prep_00.148271017.cbqs01

In jevs_aqm_prep.sh, can you remove the ecflow references `###%include <head.h>` and `###%include <envir-p1.h>`?

**Comment by Ho-ChunHuang-NOAA** [comment]
Revision done.  Log and output of "Prep" step look good to me.

**Comment by malloryprow** [comment]
Thank you!

On to the stats. Do I need to run jevs_aqm_grid2obs_stats.sh with a certain vhr?

**Comment by Ho-ChunHuang-NOAA** [comment]
Yes.  Please run the stats **hourly** as running GLOABL_ENS_CHEM from 00 to 23.

**Comment by Ho-ChunHuang-NOAA** [comment]
Leave enough time for vhr=00-22 to finish before submitting vhr=23.

**Comment by malloryprow** [comment]
I ran with vhr=00 and got two missing data emails. Should I continue? Is this expected?

> WARNING: No AQM Daily Observed file was available for 20240810
> Missing file is /lfs/h1/ops/prod/com/evs/v1.0/prep/aqm/prep/aqm/atmos.20240810/aqm/airnow_daily_20240810.nc
> Job ID: jevs_aqm_grid2obs_stats.148276809.cbqs01

> WARNING: No AQM hourly_aqobs was available for 20240809 2024080923
> Missing file is /lfs/h1/ops/prod/com/evs/v1.0/prep/aqm/prep/aqm/atmos.20240809/aqm/airnow_hourly_aqobs_2024080923.nc
> Job ID: jevs_aqm_grid2obs_stats.148276809.cbqs01

Log file is /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr513/EVS/dev/drivers/scripts/stats/aqm/jevs_aqm_grid2obs_stats.o148276809

**Comment by malloryprow** [comment]
I see now COMIN should be /lfs/h1/ops/prod/com/evs/v1.0 not /lfs/h1/ops/prod/com/evs/v1.0/prep/aqm.

**Comment by Ho-ChunHuang-NOAA** [comment]
Instruction error
Revised instruction as follow
(2) **DO NOT** Point COMIN= to /lfs/h1/ops/prod/com/evs/v1.0
Run AQM stats as it is in /dev/drivers/scripts/stats/aqm

(3) **DO NOT** Point COMIN= to /lfs/h1/ops/prod/com/evs/v1.0
Run AQM plots as it is in /dev/drivers/scripts/plots/aqm

The default is point to operational EVS output.

Sorry for the error.

**Comment by malloryprow** [comment]
Thanks, I'll get things set up accordingly. Two things:

1. Can you do a sync with develop?
2. In jevs_aqm_grid2obs_stats.sh, can you remove the ecflow references `###%include <head.h>` and `###%include <envir-p1.h>`?

**Comment by malloryprow** [comment]
I'm still a little confused about what to set to COMIN. You said the default is the operational EVS output but in jevs_aqm_grid2obs_stats.sh there is `export COMIN=/lfs/h2/emc/vpppg/noscrub/$USER/${NET}/${evs_ver_2d}`.

**Comment by Ho-ChunHuang-NOAA** [comment]
Deleted the line of "export COMIN=..." in /dev/driver, and job will use the default set in j-job.

**Comment by Ho-ChunHuang-NOAA** [comment]
If you prefer, you can set COMIN= to /lfs/h1/ops/prod/com/evs/v1.0 in stats and plots driver scripts.

**Comment by malloryprow** [comment]
Yes, I'll need to set COMIN in the driver. When deleted this is what happened

> 0.259 + compath.py prod/com/evs/v2.0.0
> Could not find evs/v2.0
> 0.845 + COMIN=''

**Comment by malloryprow** [comment]
COMOUT is /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr513/evs/v2.0
Logs are in /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr513/EVS/dev/drivers/scripts/stats/aqm

**Comment by Ho-ChunHuang-NOAA** [comment]
The log file looks good to me.

EVS PROD has not finished the hourly stats for today's AQM yet.  For daily values verification, they have the same stats values between PR testing and EVS PROD, i.e., 

evs.stats.aqm_bc.atmos.grid2obs_ozmax8.v20240810.stat  evs.stats.aqm_raw.atmos.grid2obs_ozmax8.v20240810.stat
evs.stats.aqm_bc.atmos.grid2obs_pmave.v20240810.stat   evs.stats.aqm_raw.atmos.grid2obs_pmave.v20240810.stat

**Comment by malloryprow** [comment]
Thanks! When the prod stats are finished for the hourly can you report back that they match too?

A few things before moving on to plots:

1. Can you do a sync with develop?
2. In jevs_aqm_grid2obs_stats.sh, can you remove the ecflow references `###%include <head.h>` and `###%include <envir-p1.h>`?
3. In some logs, I noticed `Cgroup memory limit exceeded`. Can you increase the memory a bit in  jevs_aqm_grid2obs_stats.sh?

**Comment by Ho-ChunHuang-NOAA** [comment]
On the second thought, please perform the plots step tomorrow so I can compare them with that from the EVS PROD.

**Comment by malloryprow** [comment]
Noticed a few things...

1. In jevs_aqm_grid2obs_plots.sh, can you remove the ecflow references ###%include <head.h> and ###%include <envir-p1.h>?
2. In all modulefiles in [https://github.com/Ho-ChunHuang-NOAA/EVS/blob/feature/EVSv2revision1/dev/modulefiles/aqm](https://github.com/Ho-ChunHuang-NOAA/EVS/blob/feature/EVSv2revision1/dev/modulefiles/aqm) change `module load PrgEnv-intel/${PrgEnv_intel_ver}` to `module load PrgEnv-intel/${PrgEnvintel_ver}`

**Comment by Ho-ChunHuang-NOAA** [comment]
Mallory:

Did you update your local branch? or correct branch?

https://github.com/Ho-ChunHuang-NOAA/EVS/blob/feature/**EVSv2revision1**/dev/drivers/scripts/plots/aqm/jevs_aqm_grid2obs_plots.sh

**Comment by malloryprow** [comment]
Ah, sorry I didn't. I see the changes now though for the plots driver.

**Comment by Ho-ChunHuang-NOAA** [comment]
It is interesting that all AQM ecf/scripts still using PrgEnv_intel_ver in EVSv1.0.10, and they are working fine ???

We should correct them in v2.0 please add the changes in the to-do list.

**Comment by malloryprow** [comment]
If they are in the ecf script, then let's fix it now too.

**Comment by malloryprow** [comment]
Thank you! For the plots, is it okay to run with VDATE=PDYm4 or should I change it to PDYm5?

**Comment by Ho-ChunHuang-NOAA** [comment]
done revision.

**Comment by Ho-ChunHuang-NOAA** [comment]
Let it run as operational without specifying VDATE, i.e., VDATE=PDYm4 in j-job.

**Comment by Ho-ChunHuang-NOAA** [comment]
I may misunderstand your words during GEFS-CHEM PR.  Is it true or false that you really want to replace DCOMIN with DCOMINobs but not separate DCOMINairnow and DCOMINaeronet?

**Comment by malloryprow** [comment]
We want to replace the DCOMIN usage with DCOMIN[obs], where the [obs] is just a placeholder for whatever the observation coming from dcomin is. The changes you did for this PR follow that, so all good.

By the way, plots job is running now and will report when it is done.

**Comment by malloryprow** [comment]
jevs_aqm_grid2obs_plots.sh run is done.

**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr513/EVS/dev/drivers/scripts/plots/aqm/jevs_aqm_grid2obs_plots.o148434617
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr513/evs/v2.0
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_aqm_grid2obs_plots.148434617.cbqs01

File: jobs/JEVS_AQM_PREP [hunk 1]
```diff
@@ -52,28 +61,18 @@ export MASKS=${MASKS:-$FIXevs/masks}
...
-export mod_ver=${mod_ver:-${aqm_ver}}
-
-
 #
 #  The AirNOW obs in DCOM will continue to be updated after first
 #  created upto 48 hours.
 #  Process the observation input only when it stop updating.
 #
-export VDATE=${VDATE:-${PDYm3}}
+export INITDATE=${INITDATE:-${PDYm3}}
```
Comment (AndrewBenjamin-NOAA) [inline jobs/JEVS_AQM_PREP:69]: I noticed both JEVS_AQM_STATS and JEVS_AQM_PLOTS both set VDATE to PDYm3. What is the purpose for changing VDATE to INITDATE in the prep jobcard?

**Review by AndrewBenjamin-NOAA — APPROVED** [review]
This PR looks good, I had one question about the change to INITDATE in the prep job, but that is for my own clarification.

File: jobs/JEVS_AQM_PREP (hunk 1 above)
Comment (Ho-ChunHuang-NOAA) [reply jobs/JEVS_AQM_PREP:69]: This is the new EVS standard and is required to be fixed as specified in AQM elements in [spreadsheet](https://docs.google.com/document/d/1vOGnCwCM3COUhQlTqHyBZrYSCPx9b6AYnxLc5NtVGEk/edit) (under AQM)

Comment (AliciaBentley-NOAA) [reply jobs/JEVS_AQM_PREP:69]: @AndrewBenjamin-NOAA @Ho-ChunHuang-NOAA To provide some additional context:

EVS prep steps gather forecasts by their INITDATE (i.e., Initialization Date) and stats/plots jobs do verification based on the VDATE (i.e., Valid Date). As an example, a prep step might grab the AQM forecasts from the 00Z, 06Z, 12Z, and 18Z initialization times. The prep step does **not** grab forecast files across 20+ initialization times that are all valid at the same VDATE. If we did things that way, forecast files may have rolled off of /dcom and /com by the time we needed them. It's better to grab an entire 12Z cycle at once, and before things roll off, so INITDATE is used in the prep step. I hope this helps!

Comment (Ho-ChunHuang-NOAA) [reply jobs/JEVS_AQM_PREP:69]: @AliciaBentley-NOAA I have no problem with the new requirement, but I want to note that AQM prep processed both OBS (by VDATE) and MOD data (by INITDATE=VDATE).  This maybe different from other components.

**Comment by Ho-ChunHuang-NOAA** [comment]
The log file looks good to me.

Spot check figures and they look good to me.  Visit [website](https://www.emc.ncep.noaa.gov/users/verification/air_quality/aqm/pr/) for more.

**Comment by malloryprow** [comment]
Great news!

I saw in https://github.com/NOAA-EMC/EVS/pull/513/commits/f3f94501dbc97390ab4c6060ef09b2a60f92ed31 you removed `export EXECevs=${EXECevs:-$HOMEevs/exec}` from the J-Jobs. Could you add it back in, please? I understand it isn't used in the aqm component, but it is one of NCO's "standard" J-Job variables and other components define it even though they aren't using it.

After that we can merge this.

**Review by malloryprow — APPROVED** [review]
Thanks for your work on this, Ho-Chun!

Don't forget to ✔️ any items this resolves on the Fixes and Additions document!

**Pull request merged**